### PR TITLE
Euler problem 001 - Optimal solution

### PR DESCRIPTION
**Pull request by nirdovrat**
Added a solution for Project Euler problem 001.
The difference is O(1) runtime complexity and a simple approach to solving it.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_001/sol8.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Multiples of 3 and 5
+
+If we list all the natural numbers below 10 that are multiples of 3 or 5,
+we get 3, 5, 6 and 9. The sum of these multiples is 23.
+
+Find the sum of all the multiples of 3 or 5 below 1000.
+"""
+
+
+def calc_arith_sum(A1: int, An: int, d: int) -> int:
```
[the remark this answers is not in the document]
Comment (nirdovrat) [reply project_euler/problem_001/sol8.py:13]: We are talking about arithmetic progression here. A1, A2.. An. I think it's more clear than a_1 and a_n, don't you think?
Also "d" is the descriptive name of this parameter - "d" of an arithmetic sequence.

Please see here:
image

File: project_euler/problem_001/sol8.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    >>> calc_arith_sum(4, 16, 4)
+    40
+    >>> calc_arith_sum(1, 8, 7)
+    9
+    """
+    num_elemens = int((An - A1) / d + 1)
+    return int(((A1 + An) * num_elemens) / 2)
+
+
+def find_An(n: int, d: int) -> int:
```
[the remark this answers is not in the document]
Comment (nirdovrat) [reply project_euler/problem_001/sol8.py:29]: same comment as above.

File: project_euler/problem_001/sol8.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    995
+    >>> find_An(1000, 15)
+    990
+    >>> find_An(1000, 10)
+    990
+    """
+    return int((n - 1) / d) * d
+
+
+def solution(n: int = 1000) -> int:
```
[the remark this answers is not in the document]
Comment (nirdovrat) [reply project_euler/problem_001/sol8.py:45]: I don't like this one either, but this is the convention in this folder (look at the other solutions there, they all have this param called "n").

**Comment by Kartik-Valluru** [comment]
@nirdovrat 
If you have read CONTRIBUTING.md , then you must have also read Coding Style, it clearly says that these codes will be seen by people all over the world and hence you need to use a coding style that is useful for even first-timers. Please make the required changes and please stop arguing with the bot.

Cheers!

**Comment by dhruvmanila** [comment]
Hello and welcome! Thanks for taking the time and effort to do but I am sorry we are not accepting any more Project Euler problems whose solutions already exists. Maybe try solving a problem which is not yet submitted?

Thanks again for doing this!

**Comment by nirdovrat** [comment]
Thanks for the feedback @dhruvmanila .
Its my first open-source contrib. so I wanted to start small :)
Totally understand though. I pushed a solution to problem 60, which seems to be unsolved so far. 
Thanks

**Pull request closed without merging**